### ISSUE TITLE
Reading list

### ISSUE THREAD

**Issue by yeun** [yeun/open-color#56]
- [x] [Contrast (Enhanced):Understanding SC 1.4.6](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast7.html)
- [x] [Reference: Use of Color](https://www.w3.org/WAI/WCAG20/quickref/#visual-audio-contrast-without-color)
- [x] [Reference: Contrast (Minimum) ](https://www.w3.org/WAI/WCAG20/quickref/#visual-audio-contrast-contrast)
- [x] [Reference: Contrast (Enhanced)](https://www.w3.org/WAI/WCAG20/quickref/#visual-audio-contrast7)
- [x] [Contrast ratio](https://www.w3.org/TR/WCAG20/#contrast-ratiodef)
- [x] [User interface component](https://www.w3.org/TR/WCAG20/#user-interface-componentdef)
- [x] [Large scale (text)](https://www.w3.org/TR/WCAG20/#larger-scaledef)
- [x] [Including a text cue for colored form control labels](https://www.w3.org/TR/WCAG20-TECHS/G205.html)
- [x] [Using a contrast ratio of 3:1 with surrounding text and providing additional visual cues on focus for links or controls where color alone is used to identify them](https://www.w3.org/TR/WCAG20-TECHS/G183.html)

**Comment by yeun** [comment]
Excerpt from (Techniques For Accessibility Evaluation And Repair Tools)[https://www.w3.org/TR/AERT#color-contrast]

# Checkpoint 2.2 - Ensure that foreground and background color combinations provide sufficient contrast when viewed by someone having color deficits or when viewed on a black and white screen

**Technique 2.2.1 [priority 3] Test the color attributes of the following elements for visibility:**

Evaluation:

- Elements:
   - BODY bgcolor | text | alink | link | vlink | background =anything  OR
   - TABLE bordercolor | bgcolor =anything  OR
   - TD | TH    bgcolor =anything OR
   - HR color =anything OR
   - any_element style="any_color_specification"
   - STYLE "any_color_specification"... STYLE
   - Where any_color_specification is defined as any CSS specification which contains:
   - color | background-color | background-image | background
- Requirement: Determine color visibility.@@needs work?

Ideally, images and multimedia object should also be tested for color visibility but algorithms are beyond the scope of this specification.

Color visibility can be determined according to the following algorithm:

**(This is a suggested algorithm that is still open to change.)**

Two colors provide good color visibility if the brightness difference and the color difference between the two colors are greater than a set range.

Color brightness is determined by the following formula:
((Red value X 299) + (Green value X 587) + (Blue value X 114)) / 1000
Note: This algorithm is taken from a formula for converting RGB values to YIQ values. This brightness value gives a perceived brightness for a color.

Color difference is determined by the following formula:
(maximum (Red value 1, Red value 2) - minimum (Red value 1, Red value 2)) + (maximum (Green value 1, Green value 2) - minimum (Green value 1, Green value 2)) + (maximum (Blue value 1, Blue value 2) - minimum (Blue value 1, Blue value 2))

The rage for color brightness difference is 125. The range for color difference is 500.

Suggested message:

Poor visibility between text and background colors.
Suggested repair:

Allow the user to change the poor color combinations.
Store any good color combinations entered by the user and use them as default prompts in the future.

**Comment by yeun** [comment]
https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast7.html

> 18 point text or 14 point bold text is judged to be large enough to require a lower contrast ratio. (See The American Printing House for the Blind Guidelines for Large Printing and The Library of Congress Guidelines for Large Print under Resources).
